src/diffusers/pipelines/controlnet/pipeline_controlnet.py

```diff
@@ -621,7 +622,36 @@ def check_inputs(
             or is_compiled
             and isinstance(self.controlnet._orig_mod, ControlNetModel)
         ):
-            self.check_image(image, prompt, prompt_embeds)
+            # When `image` is a nested list:
+            # (e.g. [[canny_image_1], [canny_image_2]])
+            if any(isinstance(i, list) for i in image):
```

Comment:
This should go to `check_image`.

Reply:
I refactored `check_inputs` and `check_image` in my latest commit.